web/src/features/time/BottomSheet.ts

```diff
@@ -0,0 +1,340 @@
+      window.removeEventListener('resize', resizeHandler);
+      resizeObserver.disconnect();
+    };
+  }
+
+  function createHandlers() {
+    let startY: number;
+    let startHeight: number;
+    let lastY: number;
+    let lastTime: number;
+    let velocity: number;
+    let isExcludedTouch: boolean;
+    let isDragging: boolean = false;
+
+    return {
+      start: (event: TouchEvent | MouseEvent) => {
+        isExcludedTouch = isExcludedElement(props, event.target);
+        if (isExcludedTouch) {
+          return;
+        }
+
+        isDragging = true;
+        startY = lastY = 'touches' in event ? event.touches[0].clientY : event.clientY;
+        startHeight = element.getBoundingClientRect().height;
+        lastTime = Date.now();
+        velocity = 0;
+        element.style.transition = 'none';
+        backdropElement.style.transition = 'none';
+
+        // Prevent text selection during drag
+        event.preventDefault();
+      },
+      move: (event: TouchEvent | MouseEvent) => {
+        if (isExcludedTouch || !isDragging) {
+          return;
+        }
+
+        const currentY = 'touches' in event ? event.touches[0].clientY : event.clientY;
+        const currentTime = Date.now();
+        const deltaY = currentY - lastY;
+        const deltaTime = currentTime - lastTime;
+
+        if (deltaTime > 0) {
+          velocity = deltaY / deltaTime;
+        }
+
+        const newHeight = calculateNewHeight(startHeight - (currentY - startY));
+        setHeight(newHeight);
+
+        lastY = currentY;
+        lastTime = currentTime;
+
+        event.preventDefault();
+      },
+      end: () => {
+        if (isExcludedTouch || !isDragging) {
+          return;
+        }
+
+        isDragging = false;
+        element.style.transition = 'height 0.3s ease-out';
+        backdropElement.style.transition = 'height 0.3s ease-out';
+        const currentHeight = element.getBoundingClientRect().height;
+
+        const targetSnap = determineTargetSnap(currentHeight, velocity);
+        setSnap(targetSnap);
+      },
+    };
+  }
+
+  function determineTargetSnap(currentHeight: number, velocity: number): number {
+    const closestSnapIndex = findClosestSnapIndex(currentHeight);
+    const velocityThreshold = 0.5; // Adjust this value to change sensitivity
+
+    if (Math.abs(velocity) > velocityThreshold) {
+      // If flicking up
+      if (velocity < 0 && closestSnapIndex < state.snapPointsWithChildHeight.length - 1) {
+        return closestSnapIndex + 1;
+      }
+      // If flicking down
+      if (velocity > 0 && closestSnapIndex > 0) {
+        return closestSnapIndex - 1;
+      }
+    }
+
+    // If velocity is low, or at the edges, snap to closest
+    return closestSnapIndex;
+  }
+
+  function calculateNewHeight(height: number): number {
+    const [lowestSnapPoint, highestSnapPoint] = [
+      Math.min(...state.snapPointsWithChildHeight),
+      Math.max(...state.snapPointsWithChildHeight),
+    ];
```

Comment:
Why use array assignemnt here instead of direct assignement to the two constants?

It will be compiled to `const lowestSnapPoint = Math.min(...state.snapPointsWithChildHeight), highestSnapPoint = Math.max(...state.snapPointsWithChildHeight)` at build anyhow (and then minified) which means using direct assignment is slightly more efficent.